### PR TITLE
Setup travis for checking builds

### PR DESCRIPTION
**Pull request by TrentHouliston**
Setup travis to check that code compiles before merging

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
-  - sudo apt-get update
+before_install:
   - sudo apt-get install puppet
-  - sudo apt-get install git
 
 script:
-  - sudo puppet apply --modulepath=puppet/modules puppet/manifests/travis.pp
-  - mkdir build
+  - sudo puppet apply --modulepath=puppet/modules puppet/manifests/nubots.pp
+  - ./b platform select native
```
Comment (JakeFountain) [inline .travis.yml]: great code Trent 👌

File: tools/decode.py [hunk 2]
```diff
@@ -47,7 +46,8 @@ def run(file, **kwargs):
         pb_type = message.__module__.split('.')[:-1]
         pb_type.append(message.__name__)
         pb_type = '.'.join(pb_type).encode('utf-8')
-        pb_hash = mmh3.hash_bytes(pb_type, 0x4e55436c, True)
+        pb_hash = '' # TODO swap to using XXHASH
```
Comment (Phasedd) [inline tools/decode.py]: This needs to be completed

Comment (TrentHouliston) [inline tools/decode.py]: A job for another PR...

**Comment by Phasedd** [comment]
The toolchain may not actually being cached from the travis output.

`FAILED: tar -Pzcf /home/travis/.casher/push.tgz /home/travis/build/NUbots/NUbots/toolchain /home/travis/.ccache`

**Pull request merged**